bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -101,6 +104,30 @@ public String getUID() {
     }
 
     /**
+     * Gets the scope of the Rule, if any exist. This property is either set by the {@link RuleEngine} when
+     * the {@link Rule} is added or by the creating party. It's an optional property.
+     * <br/><br/>Implementation note
+     *    <br/>The scope is part of the {@code UID} and the prefix thereof.
+     *    <br/>If the UID does not contain a {@link SCOPE_DELIMITER} {@code null} will be returned.
+     *    <br/>If the UID does contain a {@link SCOPE_DELIMITER} the prefix until the first occurrence will be returned.
+     *    <br/>If the prefix would have a zero length {@code null} will be returned.
+     *
+     * @return scope of this {@link Rule}, or {@code null} if no scope or an empty scope is found
+    */
+    public String getScope() {
```

Comment:
>However it would be very useful to have the getScope() within the Rule class

It will be useful if you want to divide rules into scopes i.e. in the DTAG case where you have a lot rule providers, but in case single provider the scopes can be useless.  Also in this implementation it is not possible for a rule to belong to more then one scope.
Also the uid is not required parameter when a new rule is added, there is not a problem to add a prefix for these rules, but I don't see practical reason why to do that. Why do you have to know which rules was created by specified uids and which rules are added with generated uid?
@kaikreuzer what do you think about this scope functionality of rule? Is it needed?